### PR TITLE
Add features

### PR DESCRIPTION
**Pull request by Atmos4**
Features can be:
- global: enabled for everyone
- env based: enabled for specific envs
- user based: enabled for specific users

### CODE REVIEW

File: engine/helpers.php [hunk 1]
```diff
@@ -49,6 +49,11 @@ function em(): EntityManager
     return DB::get();
 }
 
+function restrict_feature(Feature $feature, $uid = null)
```
Comment (Mathieu414) [inline engine/helpers.php:52]: That's great to restrict acces to a page from a user, but we will also need a helper function to directly get a boolean response and that could be easily user in an if clause. Like `check_auth()` but for the features, for example `check_feature(Feature::Message)`. Could be useful !

**Review by Mathieu414 — COMMENTED** [review]
Great work here, clearly a super useful thing for what comes next 😊 I was wondering, is there a way to remove access to a feature to everyone ? Like if the club upgrades his plan ;)

File: app/pages/settings/settings.php [hunk 2]
```diff
@@ -129,6 +129,8 @@
     </label>
 </form>
 
+<?= actions($is_visiting && check_auth(Access::$ROOT))->link("/licencies/$user_id/debug", "Debug", "fa-laptop-code") ?>
```
[the remark this answers is not in the document]
Comment (Atmos4) [reply app/pages/settings/settings.php]: Done, good catch!

File: app/pages/users/user_view_modal.php [hunk 3]
```diff
@@ -61,12 +61,16 @@
 
         <?php if ($can_edit_users): ?>
             <footer>
-                <nav>
+                <nav al-center>
```
[the remark this answers is not in the document]
Comment (Atmos4) [reply app/pages/users/user_view_modal.php:64]: I don't know, I wanted to try things out. Let me know if you prefer a class

File: engine/helpers.php (hunk 1 above)
Comment (Atmos4) [reply engine/helpers.php:52]: Done! I called is has_feature()

**Comment by Atmos4** [comment]
@Mathieu414 Done! Let me know if you would rather have a class than this [al-center] attribute thing :)

**Review by Mathieu414 — APPROVED** [review]
🤝

**Pull request merged**